piccolo_api/media/azure.py

```diff
@@ -0,0 +1,432 @@
+            Files are accessed via signed URLs, which are only valid for this
+            number of seconds.
+        :param upload_metadata:
+            You can provide additional metadata to the uploaded files.
+            Below we show examples of common use cases.
+
+            To set object Metadata::
+
+                AzureMediaStorage(
+                    ...,
+                    upload_metadata={'SomeMetadataTag': 'SomeMetadataValue'}
+                )
+
+            To set the content disposition (how the file behaves when opened -
+            is it downloaded, or shown in the browser)::
+
+                AzureMediaStorage(
+                    ...,
+                    # Shows the file within the browser:
+                    upload_metadata={'ContentDisposition': 'inline'}
+                )
+
+            To specify how long browsers should cache the file for::
+
+                AzureMediaStorage(
+                    ...,
+                    # Cache the file for 24 hours:
+                    upload_metadata={'CacheControl': 'max-age=86400'}
+                )
+
+            Note: We automatically add the ``ContentType`` field based on the
+            file type.
+        :param executor:
+            An executor, which file save operations are run in, to avoid
+            blocking the event loop. If not specified, we use a sensibly
+            configured :class:`ThreadPoolExecutor <concurrent.futures.ThreadPoolExecutor>`.
+        :param allowed_extensions:
+            Which file extensions are allowed. If ``None``, then all extensions
+            are allowed (not recommended unless the users are trusted).
+        :param allowed_characters:
+            Which characters are allowed in the file name. By default, it's
+            very strict. If set to ``None`` then all characters are allowed.
+        """  # noqa: E501
+        if not azure_import_successful:
+            sys.exit(
+                "Please install azure-storage-blob and azure-identity "
+                "to use this feature `pip install 'piccolo_api[azure]'`"
+            )
+
+        self.storage_account_name = storage_account_name
+        self.container_name = container_name
+        self.upload_metadata = upload_metadata or {}
+        self.folder_name = folder_name
+        self.connection_kwargs = connection_kwargs or {}
+        self.sign_urls = sign_urls
+        self.signed_url_expiry = signed_url_expiry
+        self.executor = executor or ThreadPoolExecutor(max_workers=10)
+
+        self.connection_string = None
+        self.connection_string_parts: t.Dict[str, str] = {}
+        if "connection_string" in self.connection_kwargs:
+            self.connection_string = self.connection_kwargs[
+                "connection_string"
+            ]
+        elif "AZURE_STORAGE_CONNECTION_STRING" in os.environ:
+            self.connection_string = os.environ[
+                "AZURE_STORAGE_CONNECTION_STRING"
+            ]
```

Comment:
Not really a big deal, but we can use the walrus operator to simplify this a little bit.

```python
if connection_string := (
    self.connection_kwargs.get("connection_string")
    or os.environ.get("AZURE_STORAGE_CONNECTION_STRING")
):
    self.connection_string = connection_string

```

Reply:
Managed to simplify this somewhat taking into account the previous message.